index.js

```diff
@@ -236,16 +236,22 @@ function $asString (str) {
 // magically escape strings for json
 // relying on their charCodeAt
 // everything below 32 needs JSON.stringify()
+// every string that contain surrogate needs JSON.stringify()
```

Comment:
This is not accurate. Every lone surrogate needs escaping. Surrogates that appear in valid surrogate pairs must not be escaped.

Reply:
I know @mathiasbynens 
But I read your Implementation In c. But I think implementing it in JavaScript would be heavy.
The approach of @mcollina  is an heuristic approach not deterministic.
Using of surrogate in keys of object in js is not common.
So I decided to use standard stringify when we encounter such character

Reply:
I think we should clarify that comment and write down why we use that heuristics: doing the full algorithm here would be too expensive. However the penalty to try this for lower length strings and falling back to JSON.stringify is ok.

Reply:
FWIW, I think this approach makes total sense. Lone surrogates are an edge case after all.